apps/desktop/src/lib/components/EditMode.svelte

```diff
@@ -187,6 +187,7 @@
 		modeServiceAborting = 'loading';
 
 		await modeService.abortEditAndReturnToWorkspace();
+		await modeService.awaitNotEditing();
 
```

Comment:
The main fix is simply waiting until the mode has switched before clearing the loading state.